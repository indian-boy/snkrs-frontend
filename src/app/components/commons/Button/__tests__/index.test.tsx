import { fireEvent, render } from '@testing-library/react';
import React from 'react';
import { Button } from '..';

jest.mock('react-i18next', () => ({
  useTranslation: () => {
    return {
      t: str => str,
      i18n: {
        changeLanguage: () => new Promise(() => {}),
      },
    };
  },
}));

describe('<Button  />', () => {
  it('should match snapshot', () => {
    const { getByTestId, container } = render(
      <Button data-testid="buttonID" label="Label" />,
    );

    const buttonElement = getByTestId('buttonID');

    expect(buttonElement).toBeInTheDocument();
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should match label inside the button', () => {
    const { getByTestId } = render(
      <Button data-testid="buttonID" label="Label" />,
    );

    const buttonElement = getByTestId('buttonID');
    expect(buttonElement).toHaveTextContent('Label');
  });

  it('should trigger passed onClick function', async () => {
    const onClick = jest.fn();
    const button = render(
      <Button data-testid="buttonID" label="Label" onClick={onClick} />,
    );

    fireEvent.click(button.getByText('Label'));
    expect(onClick).toHaveBeenCalled();
  });
});
