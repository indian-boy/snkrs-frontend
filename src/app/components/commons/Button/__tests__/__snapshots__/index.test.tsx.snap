// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button  /> should match snapshot 1`] = `
.c0 {
  padding: 0.9rem 1.5rem;
  border-radius: 3.125rem;
  font-style: normal;
  font-weight: bold;
  font-size: 1rem;
  cursor: pointer;
  -webkit-transition: all 0.2s ease 0s;
  transition: all 0.2s ease 0s;
  outline: none;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border: none;
  background-color: #000000;
  color: #FFFFFF;
  box-shadow: 0 0.2rem 0.5rem rgba(0,0,0,0.1);
}

.c0:focus,
.c0:active {
  outline: none;
}

.c0:hover,
.c0:focus,
.c0:active {
  background-color: #FFFFFF;
  color: #000000;
  -webkit-transform: translateY(-1px);
  -ms-transform: translateY(-1px);
  transform: translateY(-1px);
}

<button
  class="c0"
  data-testid="buttonID"
  type="button"
>
  Label
</button>
`;
